--- munin-plugins-muninexchange-20081128/plugins/mail/dkimproxy_mails	2008-11-29 13:19:37.000000000 +0100
+++ munin/plugins/mail/dkimproxy_mails	2008-11-29 13:19:43.000000000 +0100
@@ -47,7 +47,7 @@
 if [ "$1" = "config" ]; then
         echo 'graph_title DKIM Proxy mails'
         echo 'graph_order dkimnone dkimpass dkiminvalid dkimfail'
-        echo 'graph_category mail'
+        echo 'graph_category postfix'
         echo 'graph_vlabel Count'
         echo 'graph_args --base 1000 -l 0'
 #       echo 'graph_total total'
--- munin-plugins-muninexchange-20081128/plugins/mail/postfix_filtered	2008-11-29 13:19:37.000000000 +0100
+++ munin/plugins/mail/postfix_filtered	2008-11-29 13:19:43.000000000 +0100
@@ -60,7 +60,7 @@
 		else
 			echo 'graph_order allowed rbl policy helo client sender recipient relay'
 	fi
-	echo 'graph_category mail'
+	echo 'graph_category postfix'
 	echo 'graph_vlabel Mails per second'
 	echo 'graph_args --base 1000 -l 0'
 
--- munin-plugins-muninexchange-20081128/plugins/mail/postfix_mailfiltered	2008-11-29 13:19:37.000000000 +0100
+++ munin/plugins/mail/postfix_mailfiltered	2008-11-29 13:19:43.000000000 +0100
@@ -30,7 +30,7 @@
 if [ "$1" = "config" ]; then
 	echo 'graph_title Postfix daily filtering'
 	echo 'graph_order relay rbl helo client sender recipient'
-	echo 'graph_category mail'
+	echo 'graph_category postfix'
 	echo 'graph_vlabel Count'
 	echo 'graph_args --base 1000 -l 0'
 	echo 'graph_total total'
--- munin-plugins-muninexchange-20081128/plugins/mail/postfix-policyd	2008-11-29 13:19:37.000000000 +0100
+++ munin/plugins/mail/postfix-policyd	2008-11-29 13:19:43.000000000 +0100
@@ -25,7 +25,7 @@
 if [ "$1" = "config" ]; then
         echo 'graph_title Postfix-Policyd daily filtering'
         echo 'graph_order delayed passed whitelisted'
-        echo 'graph_category mail'
+        echo 'graph_category postfix'
         echo 'graph_vlabel Count'
         echo 'graph_scale no'
 
--- munin-plugins-muninexchange-20081128/plugins/mail/postfix_stats	2008-11-29 13:19:37.000000000 +0100
+++ munin/plugins/mail/postfix_stats	2008-11-29 13:19:43.000000000 +0100
@@ -19,7 +19,7 @@
 system.type COUNTER
 graph_title Postfix statistics
 graph_vlabel Postfix statistics
-graph_category Mail
+graph_category postfix
 graph_total Total
 received.label received
 delivered.label delivered
--- munin-plugins-muninexchange-20081128/plugins/mail/postgrey	2008-11-29 13:19:37.000000000 +0100
+++ munin/plugins/mail/postgrey	2008-11-29 13:19:43.000000000 +0100
@@ -30,7 +30,7 @@
 if [ "$1" = "config" ]; then
 	echo 'graph_title Postgrey daily filtering'
 	echo 'graph_order delayed passed whitelisted'
-	echo 'graph_category mail'
+	echo 'graph_category postfix'
 	echo 'graph_vlabel Count'
 	echo 'graph_scale no'
 
--- munin-plugins-muninexchange-20110326/plugins/mail/postfix_filtered_awk~	2011-03-26 15:54:42.725320884 +0100
+++ munin-plugins-muninexchange-20110326/plugins/mail/postfix_filtered_awk	2011-03-26 15:57:20.174320884 +0100
@@ -50,7 +50,7 @@
 if [ "$1" = "config" ]; then
 	echo 'graph_title Postfix message filtering'
 
-	echo 'graph_category mail'
+	echo 'graph_category postfix'
 	echo 'graph_vlabel Mails per second'
 #	echo 'graph_args --base 1000 --logarithmic'
 	echo 'graph_args --base 1000 -l 0'
--- munin-plugins-muninexchange-20110326/plugins/mail/postgrey-new~	2011-03-26 15:54:42.787320884 +0100
+++ munin-plugins-muninexchange-20110326/plugins/mail/postgrey-new	2011-03-26 15:58:43.796320879 +0100
@@ -59,7 +59,7 @@
 if(defined $ARGV[0] and $ARGV[0] eq "config") {
 	print "graph_title Postgrey Actions\n";
 	print "graph_order greylisted retry_early passed passed_w\n";
-	print "graph_category mail\n";
+	print "graph_category postfix\n";
 	print "graph_vlabel Count\n";
 	print "graph_scale no\n";
 	
