--- munin-plugins-muninexchange-20081128/plugins/ftp/vsftpd-rel~	2008-11-29 13:19:37.000000000 +0100
+++ munin-plugins-muninexchange-20081128/plugins/ftp/vsftpd-rel	2008-11-29 21:04:06.000000000 +0100
@@ -37,10 +37,10 @@
 
 install_ok()
 {
-    tempfile=$(which tempfile)
+    mktemp=$(which mktemp)
     if [ ! -r ${LOGFILE} -o \
 	 -z "${LOGTAIL}" -o ! -x "${LOGTAIL}" -o \
-	 -z "${tempfile}" -o ! -x "${tempfile}" ]
+	 -z "${mktemp}" -o ! -x "${mktemp}" ]
     then
 	return 1
     fi
@@ -50,7 +50,7 @@
 
 if [ "$1" = "autoconf" ]
 then
-    tmpfile=`which tempfile`
+    tmpfile=`which mktemp`
     if install_ok
     then
 	echo yes
@@ -82,7 +82,7 @@
 test install_ok || exit 1
 
 touch -d now-5minutes+30seconds $PIVOT
-TEMP=`tempfile`
+TEMP=`mktemp`
 trap "rm -f ${PIVOT} ${TEMP}" INT EXIT
 
 test -n "$TEMP" -a -w "$TEMP" || exit 1
--- munin-plugins-muninexchange-20081128/plugins/nagios/nagiosstatus~	2008-11-29 13:19:37.000000000 +0100
+++ munin-plugins-muninexchange-20081128/plugins/nagios/nagiosstatus	2008-11-30 01:16:55.000000000 +0100
@@ -113,10 +113,10 @@
     if($type) {
 	push(@{$values{$type}}, $_);
     }
-    if(/^host \{/) {
+    if(/^hoststatus \{/) {
 	$type = "host";
     }
-    if(/^service \{/) {
+    if(/^servicestatus \{/) {
 	$type = "service";
     }
 }
--- munin-plugins-muninexchange-20110326/plugins/mail/clamav~	2011-03-29 11:24:10.645146427 +0200
+++ munin-plugins-muninexchange-20110326/plugins/mail/clamav	2011-03-29 12:38:20.435562927 +0200
@@ -1,7 +1,8 @@
 #!/bin/sh
 #%# family=manual
 #%# capabilities=autoconf
-log=/var/log/clamav/clamd.log
+
+log=${logfile:-/var/log/clamav/clamd.log}
 
 if [ "$1" = "autoconf" ]; then
         if [ -r $log ]; then 
